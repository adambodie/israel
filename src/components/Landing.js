import React, { Component} from 'react';
import { Link } from 'react-router-dom';
import { Button } from 'react-foundation';

export default class Landing extends Component {
	constructor(){
		super();
		this.state = {
			disabled: false
		}
	}
	render() {
		return(
			<div>
				<h1>Coming Soon</h1>
				<Button className="landingLink"><Link to={'/index'} disabled={this.state.disabled} className="landingLink">Enter</Link></Button>
			</div>
		)
	}
}

