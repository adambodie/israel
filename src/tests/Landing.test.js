import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import Landing from '../components/Landing';
import {
  render,
  cleanup
} from 'react-testing-library'

afterEach(cleanup)

it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(<MemoryRouter><Landing /></MemoryRouter>, div);
	ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
	const tree = renderer.create(
    <MemoryRouter>
      <Landing />
    </MemoryRouter>
  ).toJSON();
	expect(tree).toMatchSnapshot();
});



test('loads and displays greeting', async () => {
  const { getByText } = render(<MemoryRouter><Landing/></MemoryRouter>)
	expect(getByText('Enter')).toBeInTheDocument()
})