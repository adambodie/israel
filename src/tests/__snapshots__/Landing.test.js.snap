// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="cover"
>
  <img
    alt="Cover"
    src="cover.jpg"
  />
  <a
    className="landingLink"
    href="/index"
    onClick={[Function]}
  >
    Enter
  </a>
</div>
`;
